contributors/design-proposals/dynamic-admission-control-configuration.md

```diff
@@ -0,0 +1,221 @@
+# Dynamic admission control configuration
+
+## Background
+
+[#132](https://github.com/kubernetes/community/pull/132) proposed making
+admission control extensible. In the proposal, the `initializer admission
+controller` and the `generic webhook admission controller` are the two
+controllers that set default initializers and external admission hooks for
+resources newly created. These two admission controllers are in the same binary
+as the apiserver. This [section](https://github.com/smarterclayton/community/blob/be132e88f7597ab3927b788a3de6d5ab6de673d2/contributors/design-proposals/admission_control_extension.md#dynamic-configuration)
+of #132 gave a preliminary design of the dynamic configuration of the list of
+the default admission controls. This document hashes out the implementation
+details.
+
+## Goals
+
+* Admins are able to predict what initializers/webhooks will be applied to newly
+  created objects.
+
+* Do not block the entire cluster if the intializers/webhooks are not ready
+  after registration.
+
+## Specification
+
+We assume initializers could be "fail open". We need to update #132 if this is
+accepted.
+
+The schema is copied from
+[#132](https://github.com/kubernetes/community/pull/132) with a few
+modifications.
+
+```golang
+type AdmissionControlConfiguration struct {
```

Comment:
I thought we would never trust initializers to register themselves. I thought it would be a human admin who wrote ACC. A single careless initializer could destroy the carefully ordered list of initializers, right?